apps/mobile/src/app/(home)/account/[account]/activity.tsx

```diff
@@ -0,0 +1,72 @@
+import { useAccountAddresses } from '@/hooks/use-account-addresses';
+import { useAccountActivityQuery } from '@/queries/activity/account-activity.query';
+import { Account, deserializeAccountId } from '@/store/accounts/accounts';
+import dayjs from 'dayjs';
+import { useLocalSearchParams } from 'expo-router';
+
+import { ActivityLayout } from '../../activity/activity.layout';
+import { ActivityCell } from '../../activity/components/activity-cell';
+import { ActivityIcon } from '../../activity/components/activity-icon';
+import { formatActivityType } from '../../activity/utils/format-activity';
+import { isTransfer } from '../../activity/utils/is-send-or-receive';
+import { configureAccountParamsSchema } from './index';
+
+export default function AccountActivityScreen() {
+  const params = useLocalSearchParams();
+  const { accountId } = configureAccountParamsSchema.parse(params);
+  const { fingerprint, accountIndex } = deserializeAccountId(accountId);
+  const accountAddresses = useAccountAddresses(fingerprint, accountIndex);
+  const { data: activity, isLoading } = useAccountActivityQuery(accountAddresses);
+
+  if (isLoading) {
+    return null;
+  }
+  return (
+    <ActivityLayout>
+      {activity?.map((activity, index) => {
+        const isTransferActivity = isTransfer(activity);
+        switch (activity.type) {
+          case 'sendAsset':
+          case 'receiveAsset':
+            return (
+              <ActivityCell
+                key={`activity.${index}`}
+                icon={
+                  <ActivityIcon
+                    type={activity.type}
+                    ticker={'symbol' in activity.asset ? activity.asset.symbol : ''}
+                    protocol={activity.asset.protocol}
+                  />
+                }
+                title={formatActivityType(activity.type)}
+                date={dayjs(activity.timestamp * 1000).format('MMM D, YYYY')}
+                amount={isTransferActivity ? activity.value?.crypto : undefined}
```

Comment:
This check is redundant, activity should already be narrowed down to `SendAssetActivity | ReceiveAssetActivity` within this case block, and the fallback for `value?.crypto` and `value?.fiat`is already undefined.

Reply:
I refactored this whole thing a lot, thanks for helping